src/plugins/dashboard/public/dashboard_container/embeddable/create/create_dashboard.ts

```diff
@@ -380,21 +392,45 @@ export const initializeDashboard = async ({
       ControlGroupInput,
       ControlGroupOutput,
       ControlGroupContainer
-    >(CONTROL_GROUP_TYPE);
-    const { filters, query, timeRange, viewMode, controlGroupInput, id } = initialInput;
+    >(CONTROL_GROUP_TYPE) as EmbeddableFactory<
+      ControlGroupInput,
+      ControlGroupOutput,
+      ControlGroupContainer
+    > & {
+      create: ControlGroupContainerFactory['create'];
+    };
+    const { filters, query, timeRange, viewMode, id } = initialDashboardInput;
+    const initialControlGroupInput: ControlGroupInput | {} = {
+      ...(loadDashboardReturn?.dashboardInput?.controlGroupInput ?? {}),
+      ...(sessionStorageInput?.controlGroupInput ?? {}),
+      ...(overrideInput?.controlGroupInput ?? {}),
+    };
     const fullControlGroupInput = {
       id: `control_group_${id ?? 'new_dashboard'}`,
       ...getDefaultControlGroupInput(),
-      ...pickBy(controlGroupInput, identity), // undefined keys in initialInput should not overwrite defaults
+      ...pickBy(initialControlGroupInput, identity), // undefined keys in initialInput should not overwrite defaults
       timeRange,
       viewMode,
       filters,
       query,
     };
+
     if (controlGroup) {
       controlGroup.updateInputAndReinitialize(fullControlGroupInput);
     } else {
-      const newControlGroup = await controlsGroupFactory?.create(fullControlGroupInput);
+      const lastSavedControlGroupInput = pick(
+        {
+          ...(loadDashboardReturn?.dashboardInput?.controlGroupInput ?? {}),
+          ...(overrideInput?.controlGroupInput ?? {}),
+        },
+        persistableControlGroupInputKeys
+      );
```

Comment:
I'm not entirely sure what the logic should be here.... I see, for the dashboard container, we do not take into account the `overrideInput` when setting the last saved input.